### PR TITLE
server: deprecate CRI-O userns support

### PR DESCRIPTION
**Pull request by haircommander**
as support will be beta in kubernetes soon hopefully

#### Does this PR introduce a user-facing change?

```release-note
Mark `io.kubernetes.cri-o.userns-mode`  annotation, `uid_mappings` and `gid_mappings` configuration options as deprecated. They will be replaced with Kubernetes native user namespace support in the future. 
```

### CODE REVIEW

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL

File: server/sandbox_run_linux.go [hunk 1]
```diff
@@ -426,6 +426,9 @@ func (s *Server) runPodSandbox(ctx context.Context, req *types.RunPodSandboxRequ
 	kubeAnnotations := sbox.Config().Annotations
 
 	usernsMode := kubeAnnotations[ann.UsernsModeAnnotation]
+	if usernsMode != "" {
+		log.Errorf(ctx, "Annotation 'io.kubernetes.cri-o.userns-mode' is deprecated, and will be replaced with native Kubernetes support for user namespaces in the future")
```
Comment (saschagrunert) [inline server/sandbox_run_linux.go:430]: nit: Do we want to warn here instead of error?

File: pkg/config/template.go [hunk 2]
```diff
@@ -1105,27 +1105,31 @@ const templateStringCrioRuntimeLogFilter = `# Filter the log messages by the pro
 const templateStringCrioRuntimeUIDMappings = `# The UID mappings for the user namespace of each container. A range is
 # specified in the form containerUID:HostUID:Size. Multiple ranges must be
 # separated by comma.
+# This option is deprecated, and will be replaced with native Kubernetes user namespace support in the future.
```
Comment (saschagrunert) [inline pkg/config/template.go:1108]: Do we want to include a link to the k8s.io docs here?

File: server/sandbox_run_linux.go (hunk 1 above)
Comment (haircommander) [reply server/sandbox_run_linux.go]: done!

File: pkg/config/template.go (hunk 2 above)
Comment (haircommander) [reply pkg/config/template.go]: hm how about adding the KEP number instead?:

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**